crates/parser/src/lib.rs

```diff
@@ -0,0 +1,513 @@
+use pest::{iterators::Pair, Parser};
+use pest_derive::Parser;
+
+pub mod ast;
+mod errors;
+
+use ast::{Assertion, AssertionKind, Comparison, TestDefinition};
+
+use wasm_bindgen::prelude::*;
+
+#[cfg(test)]
+mod tests;
+
+// This include forces recompiling this source file if the grammar file changes.
+// Uncomment it when doing changes to the .pest file
+const _GRAMMAR: &str = include_str!("zombienet.pest");
+
+#[derive(Parser)]
+#[grammar = "zombienet.pest"]
+pub struct ZombieNetParser;
+
+fn parse_name(pair: Pair<Rule>) -> &str {
+    return pair.into_inner().next().unwrap().as_str();
+}
+
+fn parse_within(pair: Pair<Rule>) -> u32 {
+    return pair
+        .into_inner()
+        .next()
+        .unwrap()
+        .as_str()
+        .parse::<u32>()
+        .unwrap();
+}
+
+fn parse_para_id(pair: Pair<Rule>) -> u16 {
+    return pair
+        .into_inner()
+        .next()
+        .unwrap()
+        .as_str()
+        .parse::<u16>()
+        .unwrap();
+}
+
+fn parse_taget_value(pair: Pair<Rule>) -> u64 {
+    return pair
+        .into_inner()
+        .next()
+        .unwrap()
+        .as_str()
+        .parse::<u64>()
+        .unwrap();
+}
+
+fn parse_comparison(pair: Pair<Rule>) -> ast::Comparison {
+    let mut inner_pairs = pair.into_inner();
+    let op_rule = inner_pairs.next().unwrap();
+    let op = match op_rule.as_rule() {
+        Rule::op_lte => ast::Operator::IsAtMost,
+        Rule::op_gte => ast::Operator::IsAtLeast,
+        Rule::op_lt => ast::Operator::IsBelow,
+        Rule::op_gt => ast::Operator::IsAbove,
+        Rule::op_eq => ast::Operator::Equal,
+        Rule::op_ineq => ast::Operator::NotEqual,
+        _ => {
+            println!("{:?}", op_rule);
+            unreachable!()
+        }
+    };
+
+    let target_value = inner_pairs.next().unwrap().as_str().parse::<u64>().unwrap();
+
+    ast::Comparison { op, target_value }
+}
+
+fn parse_match_pattern_rule(record: Pair<Rule>) -> (String, String, String, Option<u32>) {
```

Comment:
Contains `unwrap`s that should have explicit proof statements rather than `unwrap`s.